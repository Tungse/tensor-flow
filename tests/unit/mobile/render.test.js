import renderGalleryItems from '../../../src/js/mobile/render.js'

test('Gallery markup is rendered as expected', () => {
  const state = {
    data: {
      itemListElement: [
        {
          '@type': 'ListItem',
          'position': 1,
          'item': {
            '@type': 'ImageObject',
            'caption': '',
            'contentUrl': '//static.kino.de/wp-content/uploads/2018/05/geplante-star-wars-filme-7.jpg',
            'copyrightHolder': 'Disney',
            'copyrightYear': '2018',
            'datePublished': '2018-10-05T14:28:22',
            'width': 3500,
            'height': 1750,
            'description': 'Description 1',
            'headline': 'Headline 1',
            'url': 'https://www.example.de',
          },
        },
        {
          '@type': 'ListItem',
          'position': 2,
          'item': {
            '@type': 'SocialMediaPosting',
            'datePublished': '2018-10-05T14:28:22',
            'description': 'Description 2',
            'headline': 'Headline 2',
            'url': 'https://www.example.de/page-2',
            'sharedContent': {
              '@type': 'WebPage',
              'url': 'https://www.instagram.com/p/BcigIN3DAqE',
            },
          },
        },
        {
          '@type': 'ListItem',
          'position': 3,
          'item': {
            '@type': 'VideoObject',
            'caption': '',
            'embedUrl': 'https://videos.kino.de/embed/1143168447',
            'copyrightHolder': '',
            'copyrightYear': '',
            'datePublished': '2018-10-05T14:28:22',
            'description': 'Description 3',
            'headline': 'Headline 3',
            'thumbnail': '',
            'url': 'https://www.example.de/page-3',
            'videoFrameSize': '???',
          },
        },
      ],
    },
    currentPage: 2,
  }

  const tree = renderGalleryItems(state)
  expect(tree).toMatchSnapshot()
})
