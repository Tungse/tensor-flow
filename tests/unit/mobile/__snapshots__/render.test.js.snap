// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery markup is rendered as expected 1`] = `
"
      <div class=\\"smb-gallery-mobile\\">
      
      <div class=\\"smb-gallery-item\\">
        
          <h2>Headline 1</h2>
          <div class=\\"smb-gallery-media ImageObject\\">
            
        
          <div class=\\"embed-responsive\\" style=\\"padding-bottom: 50%\\">
            <img class=\\"embed-responsive-item lazyload\\" data-src=\\"//static.kino.de/wp-content/uploads/2018/05/geplante-star-wars-filme-7-rcm480x0u.jpg\\" alt=\\"\\">
          </div>
        
      
            <div class=\\"smb-gallery-info\\">
            
              <small>Bildquelle: Disney</small>
            
              <small>1 / undefined</small>
            </div>
          </div>
          <div class=\\"smb-gallery-content\\">
            Description 1
          </div>
          <div class=\\"smb-gallery-ed-container\\">
            <div data-slotname=\\"galleryad\\"></div>
          </div>
        </div><div class=\\"smb-gallery-item\\">
        
          <h2>Headline 2</h2>
          <div class=\\"smb-gallery-media SocialMediaPosting\\">
            <div data-role=\\"embedo\\" data-url=\\"https://www.instagram.com/p/BcigIN3DAqE\\"></div>
            <div class=\\"smb-gallery-info\\">
            <small></small>
              <small>2 / undefined</small>
            </div>
          </div>
          <div class=\\"smb-gallery-content\\">
            Description 2
          </div>
          <div class=\\"smb-gallery-ed-container\\">
            <div data-slotname=\\"galleryad2\\"></div>
          </div>
        </div><div class=\\"smb-gallery-item\\">
        
          <h2>Headline 3</h2>
          <div class=\\"smb-gallery-media VideoObject\\">
            
        
          <div class=\\"embed-responsive embed-responsive-16by9\\">
            <iframe allowfullscreen class=\\"embed-responsive-item lazyload\\" data-src=\\"https://videos.kino.de/embed/1143168447\\"></iframe>
          </div>
        
      
            <div class=\\"smb-gallery-info\\">
            <small></small>
              <small>3 / undefined</small>
            </div>
          </div>
          <div class=\\"smb-gallery-content\\">
            Description 3
          </div>
          <div class=\\"smb-gallery-ed-container\\">
            <div data-slotname=\\"galleryad3\\"></div>
          </div>
        </div>
    </div>
  "
`;
